### PR TITLE
[Method References 3] LPS-179506 journal-service

### PR DESCRIPTION
**Pull request by JanisZhang**
Convert method reference to serviceTrackerList
https://issues.liferay.com/browse/LPS-179506

### CODE REVIEW

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/JournalArticleIndexer.java [hunk 1]
```diff
@@ -151,17 +152,18 @@ public SearchContext getSearchContext() {
...
-		service = ModelDocumentContributor.class,
-		target = "(indexer.class.name=com.liferay.journal.model.JournalArticle)"
-	)
-	protected void addModelDocumentContributor(
-		ModelDocumentContributor<JournalArticle> modelDocumentContributor) {
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerList = ServiceTrackerListFactory.open(
+			bundleContext,
+			(Class<ModelDocumentContributor<JournalArticle>>)
```
Comment (dantewang) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/JournalArticleIndexer.java:159]: @JanisZhang How many `ModelDocumentContributor` trackers with different indexer.class.name filters do we have in total?

**Comment by Kvale1733** [comment]
ci:test:lima

**Comment by Kvale1733** [comment]
@JanisZhang No regressions found.

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/JournalArticleIndexer.java (hunk 1 above)
Comment (JanisZhang) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/search/JournalArticleIndexer.java:159]: Hi @dantewang,

Three classes are tracking `ModelDocumentContributor<T extends BaseModel>`:  KBArticleIndexer, JournalArticleIndexer, and WikiPageIndexer. 

All of them extend the class `BaseIndex` , but in a different module. I tried to combine them into a registry but I can not find a good way...

`JournalArticleIndexer`: this 
`KBArticleIndexer`:      https://github.com/liferay-core-infra/liferay-portal/pull/4000
`WikiPageIndexer`:      https://github.com/liferay-core-infra/liferay-portal/pull/3996

**Comment by dantewang** [comment]
ci:forward

cc @shuyangzhou @tinatian

**Comment by dantewang** [comment]
ci:forward:force

**Comment by dantewang** [comment]
ci:stop

**Pull request closed without merging**